### PR TITLE
go-counting: Implement the exercise "go-counting"

### PR DESCRIPTION
**Pull request by yunchih**
See #748

### CODE REVIEW

**Comment by yunchih** [comment]
Thanks for the head-up!  Added

File: config.json [hunk 1]
```diff
@@ -824,6 +824,18 @@
...
     {
+      "uuid": "d4ddeb18-ac22-11e7-abc4-cec278b6b50a",
+      "slug": "go-counting",
+      "core": false,
+      "unlocked_by": null,
+      "difficulty": 1,
+      "topics": [
+        "parsing",
+        "tuples",
+        "optional values"
```
Comment (m-a-ge) [inline config.json]: We've switched to `snake_case` naming convention for topics to increase consistency among tracks, so could you please choose them from the following list - https://github.com/exercism/problem-specifications/blob/master/TOPICS.txt

File: config.json [hunk 2]
```diff
@@ -824,6 +824,18 @@
       ]
     },
     {
+      "uuid": "d4ddeb18-ac22-11e7-abc4-cec278b6b50a",
+      "slug": "go-counting",
+      "core": false,
+      "unlocked_by": null,
+      "difficulty": 1,
```
Comment (m-a-ge) [inline config.json]: I guess the difficulty for this one should be increased. 
What do you think?

File: config.json (hunk 1 above)
Comment (m-a-ge) [reply config.json]: I also propose adding `classes` here

**Comment by yunchih** [comment]
Updated as requested.  Thanks for being patient on me :laughing:

**Comment by yunchih** [comment]
Hi, I've added the version string.  Anything else that needs update?

File: exercises/go-counting/README.md [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+
+To be more precise an empty intersection is part of a player's territory
+if all of its neighbours are either stones of that player or empty
+intersections that are part of that player's territory.
+
+For more information see
+[wikipedia](https://en.wikipedia.org/wiki/Go_%28game%29) or [Sensei's
+Library](http://senseis.xmp.net/).
+
+### Submitting Exercises
```
Comment (cmccandless) [inline exercises/go-counting/README.md]: RE #950, this should be `## Submitting Exercises`

File: exercises/go-counting/go_counting_test.py [hunk 4]
```diff
@@ -0,0 +1,91 @@
+import unittest
+import gocounting
+
+# Tests adapted from `problem-specifications//canonical-data.json` @ v1.1.0
```
Comment (cmccandless) [inline exercises/go-counting/go_counting_test.py]: Minor change: please add 1 additional blank line above this (2 before, 1 after) for consistency across exercises.

File: exercises/go-counting/go_counting_test.py [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+    " W W W W ",
+    "B B   B B",
+    " W BBB W ",
+    "   B B   "
+])
+
+
+class GoCountingTest(unittest.TestCase):
+    def test_5x5_for_black(self):
+        board = gocounting.GoCounting(board5x5)
```
Comment (cmccandless) [inline exercises/go-counting/go_counting_test.py]: class `GoCounting` should not be the same name as the module `gocounting`. Please rename to something more intuitive, such as `Board`

File: exercises/go-counting/go_counting_test.py [hunk 6]
```diff
@@ -0,0 +1,91 @@
...
+        stone, territory = board.territoryFor((0, 1))
+        self.assertEqual(stone, board.black)
```
Comment (cmccandless) [inline exercises/go-counting/go_counting_test.py]: It is better to define constants at either the module level
```Python
BLACK = "B"
WHITE = "W"
NONE = ""

class GoCounting(object):
    ...
```
or in a separate class
```Python
class Player(object):
    BLACK = "B"
    WHITE = "W"
    NONE = ""

class GoCounting(object):
    ...
```

**Comment by yunchih** [comment]
@cmccandless  Thanks for your review!  You've been very kind.

File: exercises/go-counting/go_counting_test.py (hunk 6 above)
Comment (cmccandless) [reply exercises/go-counting/go_counting_test.py]: You'll need to make the related changes in `go_counting_test.py` as well.

**Comment by yunchih** [comment]
@cmccandless updated as requested.

**Comment by m-a-ge** [comment]
Merged, thanks @yunchih!

**Pull request merged**